### ISSUE TITLE
Incorrect results with converted model [Caffe to Keras]

### ISSUE THREAD

**Issue by vedantbhatia** [microsoft/MMdnn#666]
Platform (like ubuntu 16.04/win10):
Ubuntu 18.04
Python version:
3.6
Source framework with version (like Tensorflow 1.4.1 with GPU):
Caffe, GPU
Destination framework with version (like CNTK 2.3 with GPU):
Keras, GPU

I have a modified GoogLeNet, in which I changed the input structure and the final number of classes. This works perfectly in Caffe. I converted this to Keras using the command:

mmconvert -sf caffe -in deploy.prototxt -iw caffe.caffemodel -df keras -om ./keras.model 

I resize the input image (grayscale) to the desired dimensions, and pass it to both keras and caffe, like so:

Caffe:
def check_keras(arr):
    arr = arr[:,None,...]
    net.blobs['data'].data[...] = arr[0]
    print(net.blobs['data'].data.shape)
    #compute
    out = net.forward()
    print(out['softmax'])
    return(np.argmax(out['softmax']))

Keras:

def check_keras(arr):
    y_train_pred_keras=m.predict(arr[...,None])
    print(y_train_pred_keras)
    return(np.argmax(y_train_pred_keras,axis=1))

For the same image, this returns different results. 

For Caffe:
[[2.8024021e-01 1.7262013e-04 7.1958715e-01]]
Predicted - class 2

For Keras:
[[0.8009005  0.00084618 0.19825332]]
Predicted - class 0

Any help is appreciated.

**Comment by rainLiuplus** [comment]
Hi @vedantbhatia, could you provide your model for us to debug?
